### DISCUSSION TITLE
[C.Deliv] Product Compositions

### DISCUSSION THREAD

**Discussion by erkist** [cdfoundation/sig-events#35] [General]
In PR https://github.com/cdfoundation/sig-events/pull/23 a Product Composition entity and corresponding event was proposed as follows:

> Announces that a new composition of source-files, artifacts, component versions and other compositions/baseline versions, has been created.
> 
> __Note:__ This event description suffers from a missing piece of terminology to represent the “recipe” by that controls what is included in new compositions. E.g. “The latest stable version (LSV) of this application, the LSV of this library, and the LSV of this configuration file”. For now, this will be referred to as a “Recipe” until a better name is given. (ESCI calls this a [Baseline](https://github.com/erkist/sig-interoperability/tree/erkist-esci/workstreams/events_in_cicd/input/enterprise-scale-ci#baseline), Eiffel likely has a term for this as well).
>
> Product Composition Created Events MUST include the following attributes:
> 
> - **Composition ID**: Unique identifier of composition.
> - **Recipe ID**: Identifier of recipe of which this is a composition.
> - **Content**: List of sources, artifacts, components and other compositions part of this composition
> 
> Optional attributes:
> 
> - **Version Name**: Machine-friendly and human-friendly version tag/number/etc. for composition, e.g. “1.4.2-alpha2”.
> - **Recipe Name**: Machine-friendly and human-friendly name of recipe, e.g. “gemeni-backbone-gen2”.
> - **Recipe Display Name**: UX/human-friendly display name of recipe, e.g. "Gemini Backbone, Gen2”.
> - **Artifacts**: List of artifacts (files, packages, Docker images etc.) part of this component version.
> - **Component Track**: States what track (e.g. platform variant (x86_64, ARM etc.), project (e.g. X-DataCenter, X-Cloud etc.) or other) the component, of which this is a version, belongs to.

As input to the discussion:

* [CompositionDefiinedEvent in Eiffel](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelCompositionDefinedEvent.md)
* [Baselines and baseline versions in ESCI](https://github.com/erkist/sig-interoperability/tree/erkist-esci/workstreams/events_in_cicd/input/enterprise-scale-ci#baseline)

Do we want to represent this concept in the spec? If so, how? What events do we want to have attached to it?

**Comment by m-linner-ericsson** [comment]
If we want to trace back from a deployment or artifact to all the commits that were part of it including 3rd party libs via our events we need the information of what other artifacts/libs a project consists of. This is then to avoid forcing the reader to look into files that will probably not be found in the same place for all projects.
